miden/src/cli/data.rs

```diff
@@ -72,18 +93,134 @@ impl InputFile {
+        match &self.advice_map {
+            Some(advice_map) => {
+                let map = advice_map
+                    .iter()
+                    .map(|(k, v)| {
+                        // decode hex key
+                        let mut key = [0u8; 32];
+                        hex::decode_to_slice(k, &mut key)
+                            .map_err(|e| format!("failed to decode advice map key `{k}` - {e}"))?;
+
+                        // convert values to Felt
+                        let values = v
+                            .iter()
+                            .map(|v| {
+                                Felt::try_from(*v).map_err(|e| {
+                                    format!(
+                                        "failed to convert advice map value `{v}` to Felt - {e}"
+                                    )
+                                })
+                            })
+                            .collect::<Result<Vec<_>, _>>()?;
+                        Ok((key, values))
+                    })
+                    .collect::<Result<HashMap<[u8; 32], Vec<Felt>>, String>>()?;
+                Ok(Some(map))
+            }
+            None => Ok(None),
+        }
+    }
+
+    /// Parse merkle store data from the input file.
+    fn parse_merkle_store(&self) -> Result<Option<MerkleStore>, String> {
+        match &self.merkle_store {
+            Some(merkle_data) => {
+                let mut merkle_store = MerkleStore::default();
+                for data in merkle_data {
+                    match data {
+                        MerkleData::MerkleTree(data) => {
+                            let leaves = Self::parse_merkle_tree(data)?;
+                            merkle_store = merkle_store.with_merkle_tree(leaves).map_err(|e| {
+                                format!("failed to add merkle tree to merkle store - {e}")
+                            })?;
+                        }
+                        MerkleData::SparseMerkleTree(data) => {
+                            let entries = Self::parse_sparse_merkle_tree(data)?;
+                            merkle_store =
+                                merkle_store.with_sparse_merkle_tree(entries).map_err(|e| {
+                                    format!(
+                                        "failed to add sparse merkle tree to merkle store - {e}"
+                                    )
+                                })?;
+                        }
+                    }
+                }
+                Ok(Some(merkle_store))
+            }
+            None => Ok(None),
+        }
+    }
+
+    /// Parse and return merkle tree leaves.
+    fn parse_merkle_tree(tree: &[String]) -> Result<Vec<Word>, String> {
+        tree.iter()
+            .map(|v| {
+                let leaf = Self::parse_leaf(v)?;
+                Ok(leaf)
+            })
+            .collect::<Result<Vec<Word>, String>>()
```

Comment:
**Suggestion:** You can likely simplify this using `try_collect` as mentioned in my other comment